src/components/JustNotSorry.js

```diff
@@ -0,0 +1,165 @@
+        }
+      });
+    }
+  }
+
+  handleContentEditableDivChange(mutations) {
+    let divCount = this.getEditableDivs().length;
+    if (divCount !== this.state.editableDivCount) {
+      this.setState({ editableDivCount: divCount });
+      if (mutations[0]) {
+        mutations.forEach((mutation) => {
+          if (
+            mutation.type === 'childList' &&
+            mutation.target.hasAttribute('contentEditable')
+          ) {
+            let id = mutation.target.id;
+            if (id) {
+              this.applyEventListeners(id);
+            }
+          }
+        });
+      }
+    }
+  }
+
+  checkForWarnings(parentElement) {
+    return Util.debounce(
+      () => this.checkForWarningsImpl(parentElement),
+      WAIT_TIME_BEFORE_RECALC_WARNINGS
+    );
+  }
+
+  checkForWarningsImpl(parentElement) {
+    this.setState({ warnings: [] });
+    this.addWarnings(parentElement);
+  }
+
+  applyEventListeners(id) {
+    let targetDiv = document.getElementById(id);
+    targetDiv.removeEventListener('focus', this.addObserver);
+    targetDiv.addEventListener('focus', this.addObserver.bind(this));
+    targetDiv.addEventListener('blur', this.removeObserver.bind(this));
+  }
+
+  addObserver(event) {
+    const element = event.currentTarget;
+    element.addEventListener(
+      'input',
+      this.checkForWarnings(element.parentNode)
+    );
+    this.addWarnings(element.parentNode);
+    this.observer.observe(element, {
+      characterData: false,
+      subtree: true,
+      childList: true,
+      attributes: false,
+    });
+  }
+
+  removeObserver(event) {
+    const element = event.currentTarget;
+    this.setState({ warnings: [] });
+    element.removeEventListener('input', this.checkForWarnings);
+    this.observer.disconnect();
+  }
+
+  getEditableDivs() {
+    return document.querySelectorAll('div[contentEditable=true]');
+  }
+
+  addWarning(node, keyword, message) {
+    const pattern = new RegExp('\\b(' + keyword + ')\\b', 'ig');
+    domRegexpMatch(node, pattern, (match, range) => {
+      let newWarning = {
+        keyword: keyword,
+        message: message,
+        parentNode: node,
+        rangeToHighlight: range,
+      };
+
+      this.setState((state) => {
+        let warnings = state.warnings;
+        warnings = state.warnings.concat(newWarning).filter(function (warning) {
+          if (warning.rangeToHighlight.startContainer) {
+            return (
+              warning.rangeToHighlight.startContainer.textContent !=
+              newWarning.message
+            );
+          }
+        });
+        return {
+          warnings,
+        };
+      });
+    });
+  }
+
+  addWarnings(node) {
+    WARNING_MESSAGES.map((warning) => {
+      this.addWarning(node, warning.keyword, warning.message);
+    });
+  }
+
+  render() {
+    const warningList = this.state.warnings.map((warning) =>
+      ReactDOM.createPortal(
+        <Warning class=".jns-warning" key={warning.keyword} value={warning} />,
```

Comment:
`class` isn't getting picked up here.  It should be `className`.  That doesn't seem to be impacting the UI, though, so maybe it's not needed at all.